core/trino-main/src/main/java/io/trino/operator/MultiChannelGroupByHash.java

```diff
@@ -819,12 +821,19 @@ public GroupByIdBlock getResult()
             extends LowCardinalityDictionaryWork<GroupByIdBlock>
     {
         private final long[] groupIds;
+        private final short[] positionToCombinationIds;
+        private final int[] combinationIdToGroupId;
+        private int nextPosition;
         private boolean finished;
 
         public GetLowCardinalityDictionaryGroupIdsWork(Page page)
         {
             super(page);
             groupIds = new long[page.getPositionCount()];
+            positionToCombinationIds = new short[page.getPositionCount()];
+            int maxCardinality = calculateMaxCardinalityAndPositionalCombinationIds(page, positionToCombinationIds);
```

Comment:
ditto: heavier computations shouldn't be part of `Work` constructor

Reply:
Done